### PR TITLE
【1.1.13】UDF function all interface added description field

### PR DESCRIPTION
**Pull request by v-kkhuang**
### What is the purpose of the change

UDF function all interface added description field

### Related issues/PRs

Related issues: #4584

### Brief change log

- Define the core abstraction and interfaces of the EngineConn Factory;
- Define the core abstraction and interfaces of Executor Manager.

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

File: linkis-public-enhancements/linkis-udf/linkis-udf-service/src/main/resources/mapper/common/UDFDao.xml [hunk 1]
```diff
@@ -114,19 +115,24 @@
...
-        from linkis_ps_udf_baseinfo
-        where create_user in
+        a.id, a.`create_user`, a.`udf_name`, a.`udf_type`,
+        a.`is_expire`,a.`is_shared`,a.`tree_id`,a.`create_time`,a.`update_time`,a.`sys`,a.`cluster_name`,b.description
+        from linkis_ps_udf_baseinfo a ,
+        (select a.* from linkis_ps_udf_version a ,
+            (SELECT udf_id ,MAX(bml_resource_version) as bml_resource_version FROM linkis_ps_udf_version GROUP BY udf_id) b
+        where b.udf_id=a.udf_id and b.bml_resource_version = a.bml_resource_version) b
+        where a.id = b.udf_id and a.create_user in
         <foreach collection="arg0" open="(" separator="," close=")" item="item">
```
Comment (casionone) [inline linkis-public-enhancements/linkis-udf/linkis-udf-service/src/main/resources/mapper/common/UDFDao.xml:125]: optimize table  alias name   not use only a simple chat （a/b/c）
Please capitalize all sql keywords
----
优化表临时别名 不要使用简单的字母 
sql 关键字请全部大写

File: linkis-public-enhancements/linkis-udf/linkis-udf-service/src/main/resources/mapper/common/UDFDao.xml [hunk 2]
```diff
@@ -114,19 +115,24 @@
...
     <select id="getSharedUDFByUser" resultMap="UDFResultMap">
         select
-        <include refid="udf_Column_List"/>
-        from linkis_ps_udf_baseinfo
-        where id in
+        a.id, a.`create_user`, a.`udf_name`, a.`udf_type`,
+        a.`is_expire`,a.`is_shared`,a.`tree_id`,a.`create_time`,a.`update_time`,a.`sys`,a.`cluster_name`,b.description
+        from linkis_ps_udf_baseinfo a , (select a.* from linkis_ps_udf_version a ,(SELECT udf_id ,MAX(bml_resource_version) as bml_resource_version FROM linkis_ps_udf_version GROUP BY udf_id) b where b.udf_id=a.udf_id and b.bml_resource_version = a.bml_resource_version) b
+        where a.id = b.udf_id and a.id in
         (select udf_id from linkis_ps_udf_shared_info where user_name=#{userName})
```
Comment (casionone) [inline linkis-public-enhancements/linkis-udf/linkis-udf-service/src/main/resources/mapper/common/UDFDao.xml:136]: optimize table  alias name   not use only a simple chat （a/b/c）
Please capitalize all sql keywords

----

优化表临时别名 不要使用简单的字母 
sql 关键字请全部大写

File: linkis-public-enhancements/linkis-udf/linkis-udf-service/src/main/resources/mapper/common/UDFDao.xml (hunk 1 above)
Comment (v-kkhuang) [reply linkis-public-enhancements/linkis-udf/linkis-udf-service/src/main/resources/mapper/common/UDFDao.xml:125]: done

File: linkis-public-enhancements/linkis-udf/linkis-udf-service/src/main/resources/mapper/common/UDFDao.xml (hunk 2 above)
Comment (v-kkhuang) [reply linkis-public-enhancements/linkis-udf/linkis-udf-service/src/main/resources/mapper/common/UDFDao.xml:136]: done

File: linkis-public-enhancements/linkis-udf/linkis-udf-service/src/main/resources/mapper/common/UDFDao.xml [hunk 3]
```diff
@@ -113,21 +114,29 @@
...
-        where create_user in
+        SELECT
+        info.id, info.`create_user`, info.`udf_name`, info.`udf_type`,
+        info.`is_expire`,info.`is_shared`,info.`tree_id`,info.`create_time`,info.`update_time`,info.`sys`,info.`cluster_name`,udf_version.description
+        FROM linkis_ps_udf_baseinfo info ,
+        (SELECT udf_version.* FROM linkis_ps_udf_version udf_version ,
+        (SELECT udf_id ,MAX(bml_resource_version) AS bml_resource_version FROM linkis_ps_udf_version GROUP BY udf_id) version_tmp
+        WHERE version_tmp.udf_id=udf_version.udf_id AND version_tmp.bml_resource_version = udf_version.bml_resource_version) udf_version
+        WHERE info.id = udf_version.udf_id AND info.create_user in
         <foreach collection="arg0" open="(" separator="," close=")" item="item">
```
Comment (casionone) [inline linkis-public-enhancements/linkis-udf/linkis-udf-service/src/main/resources/mapper/common/UDFDao.xml:125]: 优化sql 排版和使用
```

SELECT
     info.id, info.`create_user`, info.`udf_name`, info.`udf_type`,
     info.`is_expire`,info.`is_shared`,info.`tree_id`,info.`create_time`,info.`update_time`,info.`sys`,info.`cluster_name`,udf_version.description
     FROM linkis_ps_udf_baseinfo info 
INNER JOIN(     
                SELECT udfv.udf_id, udfv.description
                FROM linkis_ps_udf_version udfv 
                WHERE (udf_id,bml_resource_version) 
                        IN (
                            SELECT udf_id, MAX(bml_resource_version) bml_resource_version
                            FROM linkis_ps_udf_version
                            GROUP BY udf_id
                        ) 
) udf_version
ON udf_version.udf_id = info.id
ORDER BY id ;
```

Comment (v-kkhuang) [reply linkis-public-enhancements/linkis-udf/linkis-udf-service/src/main/resources/mapper/common/UDFDao.xml:138]: done

File: linkis-public-enhancements/linkis-udf/linkis-udf-service/src/main/resources/mapper/common/UDFDao.xml [hunk 4]
```diff
@@ -113,21 +114,29 @@
...
+        SELECT
+        info.id, info.`create_user`, info.`udf_name`, info.`udf_type`,
+        info.`is_expire`,info.`is_shared`,info.`tree_id`,info.`create_time`,info.`update_time`,info.`sys`,info.`cluster_name`,version_tmp.description
+        FROM linkis_ps_udf_baseinfo info ,
+        (SELECT version_info.* FROM linkis_ps_udf_version version_info ,
+        (SELECT udf_id ,MAX(bml_resource_version) AS bml_resource_version FROM linkis_ps_udf_version GROUP BY udf_id) version_max
+        WHERE version_max.udf_id=version_info.udf_id and version_max.bml_resource_version = version_info.bml_resource_version) version_tmp
+        WHERE info.id = version_tmp.udf_id and info.id in
+        (SELECT udf_id FROM linkis_ps_udf_shared_info WHERE user_name=#{userName})
     </select>
```
[the remark this answers is not in the document]
Comment (v-kkhuang) [reply linkis-public-enhancements/linkis-udf/linkis-udf-service/src/main/resources/mapper/common/UDFDao.xml:168]: done

**Review by casionone — APPROVED** [review]

**Pull request merged**